### PR TITLE
Add support for travis-ci integration

### PR DESCRIPTION
**Pull request by akiss77**
Fully fledged checks, builds, and tests on Linux, non-voting native builds and unit tests on OS X.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  include:
+    - os: osx
+      env: TARGET="build.darwin test-js-precommit"
+    - os: osx
+      env: TARGET=test-unit
+  allow_failures:
+    - os: osx
+
+#branches:
+#  only: master
```
Comment (LaszloLango) [inline .travis.yml]: If we don't need this, then remove it please (or uncomment if we do).

**Comment by LaszloLango** [comment]
LGTM

File: .travis.yml (hunk 1 above)
Comment (akiss77) [inline .travis.yml]: That's something to be decided. That's why I've left it there, to show that we have an option. But I can delete those lines ofc.

**Comment by akiss77** [comment]
Updated (removed the comments)

**Comment by akiss77** [comment]
Ready for landing. Does it need a squash before the merge?

**Comment by LaszloLango** [comment]
@akiss77, sure, you may land this. And yes, squash it please. Our policy is to squash all the commits into one and one commit will belongs to one PR on the master. There are a few exceptions when the patch is too huge. We can change this policy, but not now :) You could raise an issue if you think this should be changed.

**Pull request merged**